task.sql

```diff
@@ -4,30 +4,43 @@ CREATE DATABASE ShopDB;
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
-    Name VARCHAR(50),
-    PRIMARY KEY (ID)
+    ID INT PRIMARY KEY,
+    Name VARCHAR(50)
 );
 
-CREATE TABLE ProductInventory (
-    ID INT,
-    ProductName VARCHAR(50),
-    WarehouseAmount INT,
+CREATE TABLE Warehouses (
+    WarehouseID INT PRIMARY KEY,
     WarehouseName VARCHAR(50),
-    WarehouseAddress VARCHAR(50), 
+    WarehouseAddress VARCHAR(50),
     CountryID INT,
-	FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION,
-    PRIMARY KEY (ID)
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID)
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(50) UNIQUE
+);
+
+CREATE TABLE ProductInventory (
+    ID INT PRIMARY KEY,
+    ProductID INT,
+    WarehouseID INT,
+    WarehouseAmount INT,
+    FOREIGN KEY (ProductID) REFERENCES Products(ProductID),
+    FOREIGN KEY (WarehouseID) REFERENCES Warehouses(WarehouseID),
+    UNIQUE (ProductID, WarehouseID)
```

Comment:
Recommend to check if value can be repeatable

Reply:
done